scipy/stats/_sobol.pyx

```diff
@@ -406,7 +406,7 @@ cpdef void _fill_p_cumulative(cnp.float_t[::1] p,
 @cython.wraparound(False)
 cpdef void _categorize(cnp.float_t[::1] draws,
                        cnp.float_t[::1] p_cumulative,
-                       cnp.int_t[::1] result) noexcept nogil:
+                       cnp.npy_long[::1] result) noexcept nogil:
```

Comment:
Same here, either you have to override the NumPy default on the call site, or you have to unfortunately introduce:
```
ctypedef fused default_int:
    cnp.intp_t
    long
```
and use that here, to ensure that both `long` and `intp` (depending on the NumPy version) can pass.

Reply:
Alternatively: use `dtype=np.intp` and intp here.  Since that is guaranteed to be larger (if anything).